trie-root/src/lib.rs

```diff
@@ -51,16 +79,16 @@ pub trait TrieStream {
 	fn begin_branch(
 		&mut self,
 		maybe_key: Option<&[u8]>,
-		maybe_value: Option<&[u8]>,
+		maybe_value: Value,
 		has_children: impl Iterator<Item = bool>,
 	);
 	/// Append an empty child node. Optional.
 	fn append_empty_child(&mut self) {}
 	/// Wrap up a Branch node portion of a `TrieStream` and append the value
 	/// stored on the Branch (if any).
-	fn end_branch(&mut self, _value: Option<&[u8]>) {}
+	fn end_branch(&mut self, _value: Value) {}
 	/// Append a Leaf node
-	fn append_leaf(&mut self, key: &[u8], value: &[u8]);
+	fn append_leaf(&mut self, key: &[u8], value: Value);
```

Comment:
I would still prefer to have a type without NoValue here, that would prevent having to write unreachable in the TrieStream implementation in substrate.

Reply:
:+1: I did forgot about trie_root, will switch it too